bootstrap/gamma/server.go

```diff
@@ -57,33 +59,39 @@ func StartGammaServer(config GammaServerConfig) *GammaServer {
 		rootLogger, network.PublicApi(0), network.MetricRegistry(0))
 
 	s := &GammaServer{
-		ctxCancel:    cancel,
-		shutdownCond: sync.NewCond(&sync.Mutex{}),
-		httpServer:   httpServer,
-		Logger:       rootLogger,
-	}
-
-	if config.Blocking {
-		s.WaitUntilShutdown()
-	} else { // Used primarily in testing
-		go s.WaitUntilShutdown()
+		OrbsProcess: bootstrap.NewOrbsProcess(rootLogger, cancel, httpServer),
+		network:     network,
 	}
 
 	return s
 }
 
-func (n *GammaServer) GracefulShutdown(timeout time.Duration) {
-	n.ctxCancel()
-	n.httpServer.GracefulShutdown(timeout)
-	n.shutdownCond.Broadcast()
+func (n *GammaServer) Port() int {
+	return n.HttpServer.Port()
 }
 
-func (n *GammaServer) WaitUntilShutdown() {
-	n.shutdownCond.L.Lock()
-	n.shutdownCond.Wait()
-	n.shutdownCond.L.Unlock()
-}
+var (
+	port               = flag.Int("port", 8080, "The port to bind the gamma server to")
```

Comment:
Why do you think it's counter-intuitive? this is, as I understand from the docs, the *idiomatic* way to use flags. Also flags _must_ be external to the `Main()` func so that they can be set from tests